src/main/java/io/curity/identityserver/plugin/aws/authentication/CallbackGetRequestModel.java

```diff
@@ -16,28 +16,59 @@
 
 package io.curity.identityserver.plugin.aws.authentication;
 
+import se.curity.identityserver.sdk.Nullable;
 import se.curity.identityserver.sdk.web.Request;
 
-public class CallbackGetRequestModel {
-    private String _code;
-    private String _state;
-    private Request _request;
+import java.util.function.Function;
 
-    public CallbackGetRequestModel(Request request) {
-        _code = request.getParameterValueOrError("code");
-        _state = request.getParameterValueOrError("state");
-        _request = request;
+class CallbackGetRequestModel
+{
+    @Nullable
+    private final String _error;
+
+    @Nullable
+    private final String _errorDescription;
+
+    private final String _url;
+    private final String _code;
+    private final String _state;
+
+    CallbackGetRequestModel(Request request)
+    {
+        Function<String, ? extends RuntimeException> invalidParameter = (s) -> new RuntimeException(String.format(
```

Comment:
if you call the `String getQueryParameterValueOrError(String queryParameterName)` overload, this function won't be needed.

Reply:
okay